packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -287,6 +287,38 @@ export class SingleThreadedWallet
     return response.multipleChannelOutput();
   }
 
+  /**
+   * Trigger a response containing a message for counterparties, with all objectives and channels for a set of objectives.
+   * @param objectiveIds The ids of the objectives that should be sent to the counterparties
+   * @returns A promise that resolves to an object containing the messages.
+   */
+  public async syncObjectives(objectiveIds: string[]): Promise<MultipleChannelOutput> {
+    const response = WalletResponse.initialize();
+    const objectives = await this.store.getObjectivesByIds(objectiveIds);
+
+    const fetchedObjectiveIds = objectives.map(o => o.objectiveId);
+    const missingObjectives = _.difference(objectiveIds, fetchedObjectiveIds);
+
+    if (missingObjectives.length > 0) {
+      this.logger.error(
+        {objectiveIds: missingObjectives},
+        'The following objectives cannot be found'
+      );
+      throw new Error('Could not find all objectives');
+    }
+
+    for (const o of objectives) {
+      const channel = await this.store.getChannelState(o.data.targetChannelId);
+      // This will make sure any relevant channel information is synced
+      await this._syncChannel(channel.channelId, response);
+
+      const {participants} = channel;
+      response.queueCreatedObjective(o, channel.myIndex, participants);
```

Comment:
I find this confusing, since the objective wasn't just created. It might even be _incorrect_, since an app might read `newObjectives`, which is read from `response.createdObjectives` and assume that they are new objectives.

Reply:
I think this is due to my misunderstanding of the `newObjectives`.  I thought that this was communicated to other counterparties instead of the `app`. Oops!

Reply:
Fixed in d39fc70